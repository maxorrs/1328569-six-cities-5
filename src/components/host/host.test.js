import React from 'react';
import renderer from 'react-test-renderer';
import Host from './host';

describe(`Host is rendered correctly`, () => {
  it(`Host is rendered correctly with pro status`, () => {
    const props = {
      host: {
        isPro: true,
        avatarUrl: `1`,
        name: `A`
      },
      description: `123`
    };

    const tree = renderer
      .create(
          <Host {...props} />
      )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Host is rendered correctly without pro status`, () => {
    const props = {
      host: {
        isPro: false,
        avatarUrl: `1`,
        name: `A`
      },
      description: `123`
    };

    const tree = renderer
      .create(
          <Host {...props} />
      )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
