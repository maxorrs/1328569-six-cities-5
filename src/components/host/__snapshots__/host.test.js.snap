// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Host is rendered correctly Host is rendered correctly with pro status 1`] = `
<div
  className="property__host"
>
  <h2
    className="property__host-title"
  >
    Meet the host
  </h2>
  <div
    className="property__host-user user"
  >
    <div
      className="property__avatar-wrapper user__avatar-wrapper property__avatar-wrapper--pro"
    >
      <img
        alt="Host avatar"
        className="property__avatar user__avatar"
        height="74"
        src="/1"
        width="74"
      />
    </div>
    <span
      className="property__user-name"
    >
      A
    </span>
    <span
      className="property__user-status"
    >
      Pro
    </span>
  </div>
  <div
    className="property__description"
  >
    <p
      className="property__text"
    >
      123
    </p>
  </div>
</div>
`;

exports[`Host is rendered correctly Host is rendered correctly without pro status 1`] = `
<div
  className="property__host"
>
  <h2
    className="property__host-title"
  >
    Meet the host
  </h2>
  <div
    className="property__host-user user"
  >
    <div
      className="property__avatar-wrapper user__avatar-wrapper "
    >
      <img
        alt="Host avatar"
        className="property__avatar user__avatar"
        height="74"
        src="/1"
        width="74"
      />
    </div>
    <span
      className="property__user-name"
    >
      A
    </span>
  </div>
  <div
    className="property__description"
  >
    <p
      className="property__text"
    >
      123
    </p>
  </div>
</div>
`;
